### DISCUSSION TITLE
WSL中的 matplotlib 无法显示图片

### DISCUSSION THREAD

**Discussion by juehunzhe** [physics-data/faq#69] [Q&A] [answered]
WSL中无法显示图片
![QQ截图20210825175606](https://user-images.githubusercontent.com/88922481/130770254-bbc186c3-3999-4015-b5ea-986efe727552.png)

**Comment by Harry-Chen** [comment]
这是预期现象，你可以保存图片，或者在 Windows 安装 X Server，或者使用 Jupyter Notebook 等笔记本工具

**Comment by heroxbd** [comment, marked as answer]
这不正常，你的 matplotlib 是从哪里装的？使用 APT 安装的 matplotlib 都不会有这样的问题。

请给出 Python 中

```
>>> import matplotlib
>>> matplotlib.__file__
```

的输出结果，我的输出是

```
 '/usr/lib/python3/dist-packages/matplotlib/__init__.py'
```

如果不是，需要把 `.local` 目录中的文件清除，以 `apt install python3-matplotlib` 重新安装 matplotlib。

**Reply by juehunzhe** [reply]
'/home/debian/.local/lib/python3.9/site-packages/matplotlib/__init__.py'
应该就是使用apt安装的吧

**Reply by heroxbd** [reply]
`/home/debian/.local` 不是 `apt` 装的，你可能在 WSL 内用过 `pip` 安装 Python 包。

如果已经忘了曾经使用 pip 安装过包，建议将 `/home/debian/.local/` 删除。

**Comment by juehunzhe** [comment]
重新用apt安装了matplotlib，但在调用时出现问题
![QQ图片20210825225823](https://user-images.githubusercontent.com/88922481/130814778-16c2a645-f388-4347-85f5-ec17661847ca.png)

**Reply by heroxbd** [reply]
烦请在课间和答疑时间找我解决。

**Reply by heroxbd** [reply]
你没有把 `/home/debian/.local` 删干净，它还在影响着图像输出，注意倒数第8行。

传终端的出错信息时，请复制文字。

**Comment by yuanyi-thu** [comment]
我是用apt安装的，但是出现了相同的报错
![图片](https://user-images.githubusercontent.com/68137215/130892449-f7736e12-d1b8-46aa-95fa-d90eeae96c8a.png)

**Reply by heroxbd** [reply]
烦请在课间和答疑时间找我解决。
